### PR TITLE
migrate from nosetests to pytest

### PR DESCRIPTION
**Pull request by asanin-epfl**

### CODE REVIEW

File: tests/test_1_swc.py [hunk 1]
```diff
@@ -352,8 +346,8 @@ def test_neurite_wrong_root_point():
     with captured_output() as (_, err):
         with ostream_redirect(stdout=True, stderr=True):
             n = Morphology(os.path.join(_path, 'soma_cylinders.swc'))
-            assert_equal('', err.getvalue().strip())
-        assert_equal(len(n.root_sections), 1)
+            assert '' == err.getvalue().strip()
```
Comment (tomdele) [inline tests/test_1_swc.py:349]: should be :
```
assert err.getvalue().strip() == ''
```

File: tests/test_1_swc.py [hunk 2]
```diff
@@ -365,15 +359,15 @@ def test_neurite_wrong_root_point():
...
+    assert len(n.root_sections) == 2
     assert_array_equal(n.root_sections[0].points,
                        [[0,0,0], [0,0,1]])
 
     with ignored_warning(Warning.wrong_root_point):
         with captured_output() as (_, err):
             with ostream_redirect(stdout=True, stderr=True):
                 n = Morphology(os.path.join(_path, 'neurite_wrong_root_point.swc'))
-                assert_equal('', err.getvalue().strip())
+                assert '' == err.getvalue().strip()
```
Comment (tomdele) [inline tests/test_1_swc.py:370]:
```suggestion
                assert err.getvalue().strip() == ''
```

File: tests/test_9_surfaces.py [hunk 3]
```diff
@@ -12,57 +12,57 @@
...
-    #                     0.01, places=5)
+    #                     0.01, decimal=5)
 
     # # single point ASC
     # assert_almost_equal(ImmutMorphology(os.path.join(_path, "simple.asc")).soma.surface,
-    #                     4 * np.pi, places=2)
+    #                     4 * np.pi, decimal=2)
     # assert_almost_equal(Morphology(os.path.join(_path, "simple.asc")).soma.surface,
-    #                     4 * np.pi, places=2)
+    #                     4 * np.pi, decimal=2)
```
Comment (tomdele) [inline tests/test_9_surfaces.py:29]: do you need this ?

**Comment by tomdele** [comment]
Except for very minor things, it looks really nice. 

I am waiting until the CI passes

File: tests/test_9_surfaces.py (hunk 3 above)
Comment (asanin-epfl) [reply tests/test_9_surfaces.py:29]: I don't know. I just kept what was before.

File: tests/test_1_swc.py (hunk 1 above)
Comment (asanin-epfl) [reply tests/test_1_swc.py:349]: done

File: tests/test_1_swc.py (hunk 2 above)
Comment (asanin-epfl) [reply tests/test_1_swc.py:370]: done

**Comment by asanin-epfl** [comment]
> I am waiting until the CI passes

I need your help with this. I can't figure out how to make imports work correctly with pytest.  Can you please look?

**Comment by tomdele** [comment]
Can you try removing the `__init__.py` from the tests directory ?

**Comment by tomdele** [comment]
+ don t use relative imports but just : 
```
from utils import balbla
```

**Comment by asanin-epfl** [comment]
> Can you try removing the __init__.py from the tests directory ?

Thank you so much! I completely missed this file, and have been working in a different direction.

**Comment by tomdele** [comment]
Glad it helped !

**Review by tomdele — APPROVED** [review]

**Comment by asanin-epfl** [comment]
Thank you!

**Comment by tomdele** [comment]
That s really cool moving away from nose !

**Pull request merged**